django/db/models/query.py

```diff
@@ -1228,15 +1228,12 @@ def ordered(self):
             return True
         if self.query.extra_order_by or self.query.order_by:
             return True
-        elif (
+        return (
```

Comment:
Because of Python's support for [short-circuiting](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not) boolean operators, the tests are failing here as sometimes `[]` is being returned. The fix is easy:

```suggestion
        return bool(
```